homeassistant/components/unifi/config_flow.py

```diff
@@ -65,10 +76,36 @@ def async_get_options_flow(config_entry):
 
     def __init__(self):
         """Initialize the UniFi flow."""
-        self.config = None
+        self.config = {}
         self.desc = None
         self.sites = None
 
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered unifi device."""
+        parsed_url = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION])
+        friendly_name = discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME]
+        model_description = discovery_info[ssdp.ATTR_UPNP_MODEL_DESCRIPTION]
+        mac_address = format_mac(discovery_info[ssdp.ATTR_UPNP_SERIAL])
+
+        self.config = {
+            CONF_HOST: parsed_url.hostname,
+        }
+
+        if self._host_already_configured(self.config[CONF_HOST]):
+            return self.async_abort(reason="already_configured")
+
+        await self.async_set_unique_id(mac_address)
```

Comment:
I've been thinking about creating a way to set the unique id; each site has a unique ID which would be applicable and I think is a good value. But its still relevant to offer a way to disable discovery. Worst case we always need to derive the MAC the controller runs on

Reply:
Discussed on discord to keep it mac for now